spec/index.bs

```diff
@@ -1993,20 +2007,92 @@ A {{GPUBindGroup}} object has the following internal slots:
+  1. {{GPUProgrammablePassEncoder/setBindGroup(index, bindGroup, dynamicOffsets)|setBindGroup(1, ...)}}
+  1. {{GPUProgrammablePassEncoder/setBindGroup(index, bindGroup, dynamicOffsets)|setBindGroup(2, ...)}}
+  1. {{GPUComputePassEncoder/setPipeline(pipeline)|setPipeline(X)}}
+  1. {{GPUComputePassEncoder/dispatch(x, y, z)|dispatch()}}
+  1. {{GPUProgrammablePassEncoder/setBindGroup(index, bindGroup, dynamicOffsets)|setBindGroup(1, ...)}}
+  1. {{GPUComputePassEncoder/setPipeline(pipeline)|setPipeline(Y)}}
+  1. {{GPUComputePassEncoder/dispatch(x, y, z)|dispatch()}}
+
+In this scenario, the user agent would have to re-bind the group slot 2 for the second dispatch, even though neither the {{GPUBindGroupLayout}} at index 2 of {{GPUPipelineLayout/[[bindGroupLayouts]]|GPUPipelineLayout.bindGrouplayouts}}, or the {{GPUBindGroup}} at slot 2, change.
+</div>
+
+Issue: should this example and the note be moved to some "best practices" document?
+
+Note: the expected usage of the {{GPUPipelineLayout}} is placing the most common and the least frequently changing bind groups at the "bottom" of the layout, meaning lower bind group slot numbers, like 0 or 1. The more frequently a bind group needs to change between draw calls, the higher its index should be. This general guideline allows the user agent to minimize state changes between draw calls, and consequently lower the CPU overhead.
+
 ### Creation ### {#pipeline-layout-creation}
 
+A {{GPUPipelineLayout}} is created via {{GPUDevice/createPipelineLayout()|GPUDevice.createPipelineLayout()}}.
+
 <script type=idl>
 dictionary GPUPipelineLayoutDescriptor : GPUObjectDescriptorBase {
     required sequence<GPUBindGroupLayout> bindGroupLayouts;
 };
 </script>
 
+### {{GPUDevice}}.<dfn method for=GPUDevice>createPipelineLayout(descriptor)</dfn> ### {#device-create-pipeline-layout}
+
+<div algorithm=GPUDevice.createPipelineLayout>
+    **Arguments:**
+        - {{GPUPipelineLayoutDescriptor}} |descriptor|
+
+    **Returns:** {{GPUPipelineLayout}}.
+
+    1. Ensure [=pipeline layout device validation=] is not violated.
+    1. Ensure [=pipeline layout entries validation=] is not violated.
+    1. Let |pl| be a new {{GPUPipelineLayout}} object.
+    1. Set the |pl|.{{GPUPipelineLayout/[[bindGroupLayouts]]}} to |descriptor|.{{GPUPipelineLayoutDescriptor/bindGroupLayouts}}.
+    1. Return |pl|.
+
+    <div class=validusage dfn-for=GPUDevice.createPipelineLayout>
+        <dfn abstract-op>Valid Usage</dfn>
+
+        If any of the following conditions are violated:
+            1. Generate a {{GPUValidationError}} in the current scope with appropriate error message.
+            1. Create a new [=invalid=] {{GPUPipelineLayout}} and return the result.
+
+        <dfn>pipeline layout device validation</dfn>: The {{GPUDevice}} must not be lost.
+
+        <dfn>pipeline layout entries validation</dfn>:
+        There must be {{GPULimits/maxBindGroups|GPULimits.maxBindGroups}} or fewer
```

Comment:
nit: there must be device.[[limits]].maxBindGroups...

Reply:
Also I assume we'll have more limits overtime, like max dynamic uniform buffer, max textures etc?

Reply:
`device.[[limits]].maxBindGroups` - yes, I noted that in the other review myself as well. We need this change consistently across the spec. We'll follow up.

Reply:
We'll have more limits, yes. I'll add an issue to bikeshed